### ISSUE TITLE
Accessibility: Closing the product image gallery on the product single does not return users to the button that triggered it

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43606]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When closing the product gallery modal on single products (either via the close button or the escape key), keyboard focus is not returned to the large product image or magnifying glass button that triggered it. Instead, users are dumped into the footer of the website.

### Expected behavior

When a modal is closed, focus must be returned to the element that triggered it.

This ensures that keyboard-only and screen reader users don't lose their place in the page.

See these references for more information:

- [ARIA Authoring Practices Guide Dialog (Modal) Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)
- [Focus Modal Not Returned on Deque](https://docs.deque.com/issue-help/1.0.0/en/focus-modal-not-returned)

### Actual behavior

When the modal is closed, focus is lost. Hitting Tab puts the user in the header of the website.

### Steps to reproduce

1. Go to a single product and open the gallery using the magnifying glass over the product image.
2. Close the gallery using the close button or Esc key.
3. Hit Tab and observe that instead of going to the next focusable element on the page (either gallery images or the quantity selector, you are at the top of the page again.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dinhtungdu** [comment]
I tested and can confirm the bug following your reproduce steps. I added this to our backlog. Thank you for reporting this issue @amberhinds!

cc @thealexandrelara We should fix this issue as well as ensure the new Product Gallery block doesn't have the same issue.

**Comment by thealexandrelara** [comment]
Thank you for reporting this, @amberhinds!

Yes, Tung, thanks for the ping. I was testing the new Product Gallery block and noticed that it's happening there as well, so I created a new issue to track it.

**Comment by Manussakis** [comment]
I have started looking into this issue.
